### PR TITLE
Re-apply Meetup API OAuth flow code on 19.09.1

### PR DESCRIPTION
**Pull request by lucatume**
A re-application of https://github.com/moderntribe/the-events-calendar/pull/2632 on `release/B19.09.1`.

Tickets:
* https://central.tri.be/issues/130274
* https://central.tri.be/issues/125635

### CODE REVIEW

**Comment by lucatume** [comment]
This will be QAed in branch.

File: src/admin-views/aggregator/status.php [hunk 1]
```diff
@@ -176,36 +176,41 @@
...
+			$notes = esc_html__( 'The service has disabled oAuth. Some types of events may not import.', 'the-events-calendar' );
 		}
 		?>
-        <tr>
-            <td class="label">
-                <img src="<?php echo esc_url( tribe_events_resource_url( 'images/aggregator/meetup.png' ) ); ?>" /><span><?php esc_html_e( 'Meetup', 'the-events-calendar' ); ?></span>
-            </td>
-            <td class="indicator <?php esc_attr_e( $indicator ); ?>"><span class="dashicons dashicons-<?php echo esc_attr( $indicator_icons[ $indicator ] ); ?>"></span></td>
-            <td><?php echo esc_html( $text ); ?></td>
-            <td><?php echo $notes; // Escaping handled above ?></td>
```
Comment (sc0ttkclark) [inline src/admin-views/aggregator/status.php]: IS this not a useful note?

File: src/admin-views/aggregator/status.php [hunk 2]
```diff
@@ -176,36 +176,41 @@
...
-            <td><?php echo esc_html( $text ); ?></td>
-            <td><?php echo $notes; ?></td>
-        </tr>
+		<tr>
+			<td class="label">
+				<img src="<?php echo esc_url( tribe_events_resource_url( 'images/aggregator/eventbrite.png' ) ); ?>" /><span><?php esc_html_e( 'Eventbrite', 'the-events-calendar' ); ?></span>
+			</td>
+			<td class="indicator <?php esc_attr_e( $indicator ); ?>"><span class="dashicons dashicons-<?php echo esc_attr( $indicator_icons[ $indicator ] ); ?>"></span></td>
+			<td><?php echo esc_html( $text ); ?></td>
+			<td><?php echo $notes; // Escaping handled above. ?></td>
```
Comment (sc0ttkclark) [inline src/admin-views/aggregator/status.php]: Is this not a useful note?

File: src/admin-views/aggregator/status.php [hunk 3]
```diff
@@ -77,7 +77,7 @@
 			<td class="label"><?php esc_html_e( 'Current usage', 'the-events-calendar' ); ?></td>
 			<td class="indicator <?php esc_attr_e( $indicator ); ?>"><span class="dashicons dashicons-<?php echo esc_attr( $indicator_icons[ $indicator ] ); ?>"></span></td>
 			<td><?php echo esc_html( $text ); ?></td>
-			<td><?php echo $notes; ?></td>
+			<td><?php echo $notes;  // Escaping handled above. ?></td>
```
Comment (sc0ttkclark) [inline src/admin-views/aggregator/status.php]: Is this not a useful note?

File: src/Tribe/Aggregator/Record/Meetup.php [hunk 4]
```diff
@@ -77,4 +77,55 @@ public static function filter_event_to_preserve_fields( $event, $record ) {
...
+	 * @since TBD
+	 *
+	 * @param array $args
+	 *
+	 * @return string Either the URL to obtain Eventbrite authorization token or an empty string.
+	 */
+	public static function get_auth_url( $args = array() ) {
+		$service = tribe( 'events-aggregator.service' );
+
+		if ( $service->api() instanceof WP_Error ) {
```
Comment (sc0ttkclark) [inline src/Tribe/Aggregator/Record/Meetup.php]: Minor tweak, but I suggest we set `$api` above this line and reference that here, and remove it from being set further below.

File: src/Tribe/Aggregator/Record/Meetup.php [hunk 5]
```diff
@@ -77,4 +77,55 @@ public static function filter_event_to_preserve_fields( $event, $record ) {
...
+			$meetup_license = $api->licenses['tribe-meetup'];
+
+			if ( empty( $key ) ) {
+				$key = $meetup_license;
+			} else {
+				$key2 = $meetup_license;
+			}
+		}
+
+		$url = $service->api()->domain . 'meetup/' . $key;
```
Comment (sc0ttkclark) [inline src/Tribe/Aggregator/Record/Meetup.php]:
```suggestion
		$url = $api->domain . 'meetup/' . $key;
```

File: src/admin-views/aggregator/origins/meetup.php [hunk 6]
```diff
@@ -15,65 +15,42 @@
...
-		?>
-	</td>
-</tr>
-<tr class="tribe-dependent" data-depends="#tribe-ea-field-origin" data-condition="meetup">
-	<th scope="row" class="tribe-dependent" data-depends="#tribe-has-meetup-credentials" data-condition="1">
+		</td>
+	</tr>
+<?php endif; ?>
+
+<tr class="tribe-dependent" data-depends="<?php echo esc_attr( $data_depends ) ?>"
```
Comment (sc0ttkclark) [inline src/admin-views/aggregator/origins/meetup.php]:
```suggestion
<tr class="tribe-dependent" data-depends="<?php echo esc_attr( $data_depends );
 ?>"
```

File: src/admin-views/aggregator/origins/meetup.php [hunk 7]
```diff
@@ -15,65 +15,42 @@
...
+    data-condition="<?php echo esc_attr( $data_condition ) ?>">
```
Comment (sc0ttkclark) [inline src/admin-views/aggregator/origins/meetup.php]:
```suggestion
    data-condition="<?php echo esc_attr( $data_condition ); ?>">
```

**Comment by lucatume** [comment]
Thanks for applying the changes @sc0ttkclark

File: src/admin-views/aggregator/status.php (hunk 1 above)
Comment (lucatume) [reply src/admin-views/aggregator/status.php]: It is, no idea why I removed it.

File: src/Tribe/Aggregator/Record/Meetup.php (hunk 4 above)
Comment (lucatume) [reply src/Tribe/Aggregator/Record/Meetup.php]: Yep, better, thanks.

File: src/Tribe/Aggregator/API/Origins.php [hunk 8]
```diff
@@ -176,56 +181,40 @@ private function fetch_origin_data() {
...
 
 	/**
-	 * Get origin limit values
+	 * Get origin limit values for an operation.
 	 *
-	 * @param string $type Type of limit to retrieve
+	 * @param string $type Type of operation limit to retrieve; defaults to `import`.
 	 *
-	 * @return int
+	 * @return int The limit applied to t
```
Comment (bordoni) [inline src/Tribe/Aggregator/API/Origins.php]: Missing the end of this line.

File: src/Tribe/Aggregator/API/Origins.php [hunk 9]
```diff
@@ -4,6 +4,8 @@
 
 class Tribe__Events__Aggregator__API__Origins extends Tribe__Events__Aggregator__API__Abstract {
 
+	const VERSION = '1.1.0';
```
Comment (bordoni) [inline src/Tribe/Aggregator/API/Origins.php]: Since?

File: src/Tribe/Aggregator/Settings.php [hunk 10]
```diff
@@ -175,20 +176,175 @@ public function is_ea_authorized_for_eb() {
...
+		}
+
+		if ( ! $this->handle_meetup_security_key( $meetup_authorized ) ) {
+			return false;
+		}
+
+		return true;
+	}
+	/*
+		===========================
```
Comment (bordoni) [inline src/Tribe/Aggregator/Settings.php]: Should we remove this?

**Pull request merged**